### PR TITLE
Feat: Improve text line wrapper

### PR DESCRIPTION
**Pull request by borisevich-a-v**
1) Fix text line wrapping
2) Fix white spaces at the beginning of a line

### CODE REVIEW

**Comment by borisevich-a-v** [comment]
It works better now
image

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py [hunk 1]
```diff
@@ -48,11 +107,6 @@ def convert(self, text: str) -> Page:
         page_size = self.calculate_page_size(tokens, self.page_border_offset)
         return Page(page_num=1, size=page_size, objs=tokens)
 
-    def _get_paragraphs(self, text: str) -> List[str]:
```
Comment (borisevich-a-v) [inline convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:51]: Replaced with `text.splitlines(keepends=True)`

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py [hunk 2]
```diff
@@ -31,14 +88,15 @@ def __init__(
         self.font_width = font_width
         self.line_spacing = line_spacing
 
-    def convert(self, text: str) -> Page:
-        paragraphs = self._get_paragraphs(text)
+        self.text_wrapper = TextWrapper(line_length=self.line_tokens_amount)
 
+    def convert(self, text: str) -> Page:
+        paragraphs = text.splitlines(keepends=True)
```
Comment (serereg) [inline convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:94]: Do we need these paragraphs, while TextWrapper can work with "\n"?

File: convert/tests/test_vertex/test_text_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import collections
+import string
+
+from src.vertex_to_bd.badgerdoc_format.plain_text_converter import TextWrapper
+
+
+def test_pop_beginning_whitespaces_text_begin_with_whitespaces():
```
Comment (andgineer) [inline convert/tests/test_vertex/test_text_wrapper.py:7]: I would also add tests string with more that one space char at the beginning

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py [hunk 4]
```diff
@@ -10,8 +12,63 @@
...
+        while text:
+            if len(line) >= self.line_length:
+                lines.append(line)
+                line = []
+
+            word = self.pop_next_word(text)
+
+            if len(word) + len(line) <= self.line_length:
+                line.extend(word)
+            else:
```
Comment (andgineer) [inline convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:59]: this way with first long word you will add first empty line

Comment (andgineer) [reply convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:59]: just add the test
        tw = TextWrapper(line_length=10)
        text = "very_long_word word1 word2"

and fix the bug

File: convert/tests/test_vertex/test_text_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    tw = TextWrapper(line_length=40)
+    text = "word1 word2 word3 word4 word5"
+
+    result = tw.wrap(text)
+
+    assert result == [text]
+
+
+def test_wrap_text_started_with_spaces():
+    tw = TextWrapper(line_length=20)
```
Comment (andgineer) [inline convert/tests/test_vertex/test_text_wrapper.py:53]: please name the magic number

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py [hunk 6]
```diff
@@ -10,8 +12,63 @@
...
 
-def generate_chunks(text: str, size: int = 70) -> List[str]:
-    return [text[i : i + size] for i in range(0, len(text), size)]
+def generate_chunks(obj_to_split: List[str], size: int) -> List[List[str]]:
+    return [
+        obj_to_split[i : i + size] for i in range(0, len(obj_to_split), size)
+    ]
+
+
+class TextWrapper:
```
[the remark this answers is not in the document]
Comment (borisevich-a-v) [reply convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:21]: Can you offer a name?

File: convert/tests/test_vertex/test_text_wrapper.py (hunk 3 above)
Comment (borisevich-a-v) [reply convert/tests/test_vertex/test_text_wrapper.py:7]: Sorry, but this test uses text with several whitespaces at the beginning

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py (hunk 4 above)
Comment (borisevich-a-v) [reply convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py:59]: fixed

File: convert/tests/test_vertex/test_text_wrapper.py (hunk 5 above)
Comment (borisevich-a-v) [reply convert/tests/test_vertex/test_text_wrapper.py:53]: I've removed the numbers from expected result output

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py [hunk 7]
```diff
@@ -10,8 +12,63 @@
...
+        return wsp
+
+    def pop_next_word(self, text: Deque[str]) -> List[str]:
+        word = []
+        while text and text[0] not in string.whitespace:
+            word.append(text.popleft())
+        return word
+
+    def wrap(self, text: str) -> List[str]:
+        text = collections.deque(text)
```
[the remark this answers is not in the document]
Comment (borisevich-a-v) [reply convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py]: renamed

File: convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py (hunk 2 above)
Comment (borisevich-a-v) [reply convert/src/vertex_to_bd/badgerdoc_format/plain_text_converter.py]: Good point! Fixed

**Review by serereg — APPROVED** [review]

**Review by spiridonovfed — APPROVED** [review]

**Pull request merged**